Mathlib/RingTheory/AdicCompletion/Basic.lean

```diff
@@ -201,67 +210,204 @@ instance (priority := 100) of_subsingleton [Subsingleton M] : IsPrecomplete I M
   toFun f := f.1 n
   map_add' _ _ := rfl
   map_smul' _ _ := rfl
-#align adic_completion.eval adicCompletion.eval
+#align adic_completion.eval AdicCompletion.eval
 
 @[simp]
 theorem coe_eval (n : ℕ) :
-    (eval I M n : adicCompletion I M → M ⧸ (I ^ n • ⊤ : Submodule R M)) = fun f => f.1 n :=
+    (eval I M n : AdicCompletion I M → M ⧸ (I ^ n • ⊤ : Submodule R M)) = fun f => f.1 n :=
   rfl
-#align adic_completion.coe_eval adicCompletion.coe_eval
+#align adic_completion.coe_eval AdicCompletion.coe_eval
 
-theorem eval_apply (n : ℕ) (f : adicCompletion I M) : eval I M n f = f.1 n :=
+theorem eval_apply (n : ℕ) (f : AdicCompletion I M) : eval I M n f = f.1 n :=
   rfl
-#align adic_completion.eval_apply adicCompletion.eval_apply
+#align adic_completion.eval_apply AdicCompletion.eval_apply
 
-theorem eval_of (n : ℕ) (x : M) : eval I M n (of I M x) = mkQ _ x :=
+theorem eval_of (n : ℕ) (x : M) : eval I M n (of I M x) = mkQ (I ^ n • ⊤ : Submodule R M) x :=
   rfl
-#align adic_completion.eval_of adicCompletion.eval_of
+#align adic_completion.eval_of AdicCompletion.eval_of
 
 @[simp]
 theorem eval_comp_of (n : ℕ) : (eval I M n).comp (of I M) = mkQ _ :=
   rfl
-#align adic_completion.eval_comp_of adicCompletion.eval_comp_of
+#align adic_completion.eval_comp_of AdicCompletion.eval_comp_of
 
 @[simp]
 theorem range_eval (n : ℕ) : LinearMap.range (eval I M n) = ⊤ :=
   LinearMap.range_eq_top.2 fun x => Quotient.inductionOn' x fun x => ⟨of I M x, rfl⟩
-#align adic_completion.range_eval adicCompletion.range_eval
+#align adic_completion.range_eval AdicCompletion.range_eval
 
 variable {I M}
 
 @[ext]
-theorem ext {x y : adicCompletion I M} (h : ∀ n, eval I M n x = eval I M n y) : x = y :=
+theorem ext {x y : AdicCompletion I M} (h : ∀ n, eval I M n x = eval I M n y) : x = y :=
   Subtype.eq <| funext h
-#align adic_completion.ext adicCompletion.ext
+#align adic_completion.ext AdicCompletion.ext
 
 variable (I M)
 
-instance : IsHausdorff I (adicCompletion I M) :=
+instance : IsHausdorff I (AdicCompletion I M) :=
   ⟨fun x hx => ext fun n => smul_induction_on (SModEq.zero.1 <| hx n) (fun r hr x _ =>
     ((eval I M n).map_smul r x).symm ▸
       Quotient.inductionOn' (eval I M n x) fun x => SModEq.zero.2 <| smul_mem_smul hr mem_top)
     fun _ _ ih1 ih2 => by rw [LinearMap.map_add, ih1, ih2, LinearMap.map_zero, add_zero]⟩
 
-end adicCompletion
+@[simp]
+theorem transitionMap_mk {m n : ℕ} (hmn : m ≤ n) (x : M) :
+    transitionMap I M hmn
+      (Submodule.Quotient.mk (p := (I ^ n • ⊤ : Submodule R M)) x) =
+      Submodule.Quotient.mk (p := (I ^ m • ⊤ : Submodule R M)) x := by
+  simp [transitionMap]
+
+@[simp]
+theorem transitionMap_eq (n : ℕ) : transitionMap I M (Nat.le_refl n) = LinearMap.id := by
+  ext
+  simp
+
+@[simp]
+theorem transitionMap_comp {m n k : ℕ} (hmn : m ≤ n) (hnk : n ≤ k) :
+    transitionMap I M hmn ∘ₗ transitionMap I M hnk = transitionMap I M (hmn.trans hnk) := by
```

Comment:
Can you add `(transitionMap I M hmn (transitionMap I M hnk x)) = transitionMap I M (hmn.trans hnk) x` as a simp lemma?